### PR TITLE
Use ctypes for libnetfilter_queue

### PR DESCRIPTION
**Pull request by brian-goldman**
This removes a dependency and a compile step.

### CODE REVIEW

File: packet_queue/libnetfilter_queue.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+class Manager(object):
+  """Manages multiple queues."""
+
+  def __init__(self):
+    self.handle = nfq.nfq_open()
+    self.fileno = nfq.nfq_fd(self.handle)
+    self.socket = socket.fromfd(self.fileno, socket.AF_UNIX, socket.SOCK_RAW)
+
+    nfq.nfq_get_msg_packet_hdr.restype = ctypes.POINTER(msg_packet_header)
+    nfq.nfq_get_msg_packet_hdr.argtypes = ctypes.POINTER(nfq_data),
```
Comment (brian-goldman) [inline packet_queue/libnetfilter_queue.py]: This should be pulled up to the module level. I'll fix it right now.

File: README.md [hunk 2]
```diff
@@ -13,9 +13,8 @@ abstracting the concept of a packet from IP and UDP.
 ## Installation
 
 ```
-apt-get install libnetfilter-queue-dev
```
Comment (minkezhang) [inline README.md]: This is still needed, else 'libnetfilter_queue.so' is not found

Comment (brian-goldman) [inline README.md]: Fixed.

File: packet_queue/libnetfilter_queue.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    """Set the verdict on a Packet instance: NF_ACCEPT or NF_DROP."""
+    payload_buffer = ctypes.create_string_buffer(packet.payload)
+    nfq.nfq_set_verdict(packet.qh,
+                        packet.id,
+                        verdict,
+                        packet.size,
+                        payload_buffer)
+
+  def bind(self, queue_num, callback):
+    """Bind a queue number to a callback and return its file descriptor.
```
Comment (brian-goldman) [inline packet_queue/libnetfilter_queue.py]: Just noticed that this doesn't actually return anything. Doesn't need to. I'll fix the doc now.

File: packet_queue/libnetfilter_queue.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  packet = Packet(packet_id, size, payload, qh)
+  py_callbacks[qh](packet)
+  return 0
+
+# Maps queue handles to user-specified callbacks.
+py_callbacks = {}
+
+
+class Manager(object):
+  """Manages multiple queues."""
```
Comment (minkezhang) [inline packet_queue/libnetfilter_queue.py:60]: I'm not sure where "multiple" comes in -- we're only ever opening one handle per manager. Is this a feature we would like to implement in the future?

File: packet_queue/nfqueue.py [hunk 5]
```diff
@@ -48,31 +52,33 @@ def configure(protocol, port, pipes, interface):
       raise ValueError("Given interface does not exist.", interface)
 
   add(protocol, port, interface)
-  queue = netfilterqueue.NetfilterQueue()
+  manager = libnetfilter_queue.Manager()
```
Comment (minkezhang) [inline packet_queue/nfqueue.py:55]: I'm not sure I understand what the point of this instantiation is -- we aren't returning this instance, and it's not being used directly to set self.manager. (What did queue = ... originally do?)

Comment (minkezhang) [inline packet_queue/nfqueue.py:55]: Never mind I missed the indentation in the following lines.

File: packet_queue/libnetfilter_queue.py (hunk 4 above)
Comment (minkezhang) [inline packet_queue/libnetfilter_queue.py:60]: Never mind.

Comment (brian-goldman) [inline packet_queue/libnetfilter_queue.py:60]: Yeah, we _used to_ only handle one queue, with header inspection to determine what direction each packet was heading. The future is now.

**Pull request merged**